boxes/generators/agricolainsert.py

```diff
@@ -0,0 +1,882 @@
+                ],
+                move="up",
+            )
+        self.rectangularWall(length, height / 2, "ffef", move="up")
+
+        for i in range(2):
+            self.rectangularWall(
+                2 * row_width + self.thickness,
+                lowered_height,
+                [
+                    "F",
+                    "F",
+                    MoorBoxHoleEdge(self, height, lowered_corner_height, i % 2 == 0),
+                    "F",
+                ],
+                move="up",
+                callback=[
+                    partial(self.generate_side_finger_holes, row_width, height / 2)
+                ],
+            )
+
+        self.ctx.restore()
+        self.rectangularWall(length, 0, "FFFF", move="right only")
+
+    def generate_side_finger_holes(self, row_width, height):
+        self.fingerHolesAt(row_width + 0.5 * self.thickness, 0, height)
+
+    def render_room_boxes(self, height, row_width):
+        """
+        The two boxes for field/wood room tiles and clay/stone room tiles.
+        """
+        wood_room_box_width = 39.8
+        self.render_room_box(wood_room_box_width, height, row_width)
+        stone_room_box_width = 26.7
+        self.render_room_box(stone_room_box_width, height, row_width)
+
+    def render_room_box(self, width, height, row_width):
+        """
+        A box in which storing room/field tiles.
+        """
+        border_height = 12
+        room_box_length = row_width * 2 + self.thickness
+
+        self.ctx.save()
+
+        self.rectangularWall(
+            room_box_length,
+            height,
+            "eFfF",
+            move="up",
+            callback=[partial(self.generate_side_finger_holes, row_width, height)],
+        )
+
+        self.rectangularWall(
+            room_box_length,
+            width,
+            "FFfF",
+            move="up",
+            callback=[partial(self.generate_side_finger_holes, row_width, width)],
+        )
+
+        self.rectangularWall(
+            room_box_length,
+            border_height,
+            "FFeF",
+            move="up",
+            callback=[
+                partial(self.generate_side_finger_holes, row_width, border_height)
+            ],
+        )
+
+        for _ in range(3):
+            self.trapezoidWall(width, height, border_height, "ffef", move="up")
+
+        self.ctx.restore()
+
+        self.rectangularWall(room_box_length, 0, "FFFF", move="right only")
+
+    def render_player_box(self, player_box_height):
+        """
+        A box in which storing all the bits of a single player,
+        including (and designed for) the cardboard bed from Farmers Of The Moor.
+        """
+        self.ctx.save()
+        bed_inner_height = player_box_height - self.thickness
+        bed_inner_length = 66.75
+        bed_inner_width = 50.5
+        bed_foot_height = 6.5
+
+        bed_edge = Bed2SidesEdge(self, bed_inner_length)
+        noop_edge = NoopEdge(self)
+        for _ in range(2):
```

Comment:
With a bit of moveTo() magic these could probably be placed a bit better by turning the second part 180°.

Reply:
I totally agree, in fact I actually cut them like you are suggesting :-) But most of the time I don't use boxespy for layout optimization. I prefer to optimize more aggressively directly in Inkscape, and also take into account the shape of the piece of plywood I use.
I find that using moveTo adds complexity in this case, I did a separate commit for that.